Added function to get DM device subsystem

japokorn opened the pull request:
added function to get DM device subsystem using device mapper
added tests for this function

vojtechtrefny commented on src/plugins/dm.c:
```diff
@@ -265,6 +265,27 @@ gchar* bd_dm_node_from_name (const gchar *map_name, GError **error) {
+ *
+ * Returns: subsystem of the given device
+ */
+gchar* bd_dm_get_subsystem_from_name (const gchar *device_name, GError **error) {
```
Please add this function to ` docs/libblockdev-sections.txt` too.

japokorn replied:
Fixed.

japokorn commented:
Changed commit message as well as PR name and description to be more specific.
Modified test to use LVM tools for creation of DM device to improve the test value.

vpodzime commented on tests/dm_test.py:
```diff
@@ -39,6 +39,21 @@ def _clean_up(self):
+    def test_get_subsystem_from_name(self):
+        """Verify that it is possible to get device subsystem from its name"""
+        run("pvcreate -ff -y %s >/dev/null 2>&1" % self.loop_dev)
+        run("vgcreate -s10M libbd_dm_tests %s >/dev/null 2>&1" % self.loop_dev)
```
These two steps can be combined into a single one -- just the ``vgcreate``. And is there any reason to specify the extent size?

japokorn replied:
Fixed. Extent size removed.

vpodzime commented on src/lib/plugin_apis/dm.api:
```diff
@@ -88,6 +88,15 @@ gchar* bd_dm_name_from_node (const gchar *dm_node, GError **error);
+ * @device_name: name of the device
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: subsystem of the given device
```
Please also specify ``Tech category: %BD_DM_TECH_RAID-%BD_DM_TECH_QUERY`` here (and in the *.c* file).

japokorn replied:
Fixed.

vpodzime commented on tests/dm_test.py:
```diff
@@ -39,6 +39,21 @@ def _clean_up(self):
+        run("pvcreate -ff -y %s >/dev/null 2>&1" % self.loop_dev)
+        run("vgcreate -s10M libbd_dm_tests %s >/dev/null 2>&1" % self.loop_dev)
+        run("lvcreate -n subsystem_test -L50M libbd_dm_tests >/dev/null 2>&1")
+        self.addCleanup(self._destroy_lvm)
```
Please add the cleanup first, that way it will run even if the creation fails half way.

japokorn replied:
Fixed.

vpodzime commented on tests/dm_test.py:
```diff
@@ -39,6 +39,21 @@ def _clean_up(self):
+        self.addCleanup(self._destroy_lvm)
+
+        subsystem = BlockDev.dm_get_subsystem_from_name("libbd_dm_tests-subsystem_test")
+        self.assertEqual(subsystem, "LVM")
```
Would be great to also have a test for some other subsystem. The other easiest one is probably LUKS (crypt).

japokorn replied:
Fixed. I have written test for luks device. Since it requires reinitialization of the resources used in LVM test, I have put it into a new function.

vojtechtrefny commented on tests/dm_test.py:
```diff
@@ -39,6 +39,34 @@ def _clean_up(self):
+    def test_get_subsystem_from_name_crypt(self):
+        """Verify that it is possible to get luks device subsystem from its name"""
+        self.addCleanup(self._destroy_crypt)
+        self.addCleanup(self._destroy_lvm)
```
Is this needed for this test?

japokorn replied:
No. Fixed.

vojtechtrefny commented on tests/dm_test.py:
```diff
@@ -39,6 +39,34 @@ def _clean_up(self):
+
+    def _destroy_crypt(self):
+        run("cryptsetup close libbd_dm_tests-subsystem_crypt")
+                #>/dev/null 2>&1")
```
This looks like something you forgot to delete :)

japokorn replied:
Indeed I did. Fixed.

japokorn commented:
jenkins, test this please

japokorn commented:
jenkins, test this please

japokorn commented:
Pushed.
